struqture/src/fermions/fermionic_indices.rs

```diff
@@ -1224,6 +1228,149 @@ fn sort_and_signal(indices: TinyVec<[usize; 2]>) -> (TinyVec<[usize; 2]>, bool,
+    ///
+    /// # Returns
+    ///
+    /// `SpinHamiltonian` - The spin operator that results from the transformation.
+    ///
+    /// * Unexpectedly failed to add a PauliProduct to a SpinOperator or SpinHamiltonian internal struqture bug.
+    ///
+    /// # Panics
+    ///
+    /// * Internal bug in `add_operator_product`.
+    fn jordan_wigner(&self) -> Self::Output {
+        let number_creators = self.number_creators();
+        let number_annihilators = self.number_annihilators();
+        let mut spin_operator = SpinOperator::new();
+
+        let mut id = PauliProduct::new();
+        id = id.set_pauli(0, SingleSpinOperator::Identity);
+        spin_operator
+            .add_operator_product(id, CalculatorComplex::new(1.0, 0.0))
+            .expect("Internal bug in add_operator_product.");
+
+        // Jordan-Wigner strings are inserted every second lowering (raising) operator, in even or
+        // odd positions depending on the parity of the total number of creation (annihilation)
+        // operators.
+        let mut previous = 0;
+        for (index, site) in self.creators().enumerate() {
+            if index % 2 != number_creators % 2 {
+                for i in previous..*site {
+                    spin_operator = spin_operator * PauliProduct::new().z(i)
+                }
+            }
+            spin_operator = spin_operator * _lowering_operator(site);
+            previous = *site;
+        }
+
+        previous = 0;
+        for (index, site) in self.annihilators().enumerate() {
+            if index % 2 != number_annihilators % 2 {
+                for i in previous..*site {
+                    spin_operator = spin_operator * PauliProduct::new().z(i)
+                }
+            }
+            spin_operator = spin_operator * _raising_operator(site);
+            previous = *site;
+        }
+
+        // Spin terms with imaginary coefficients are dropped, and
+        // real coefficients are doubled.
+        if !self.is_natural_hermitian() {
+            let mut out = SpinHamiltonian::new();
+            for (product, coeff) in spin_operator.iter() {
+                if coeff.im == 0.0.into() {
+                    out.add_operator_product(product.clone(), coeff.re.clone() * 2)
+                        .expect("Internal bug in add_operator_product.");
+                }
+            }
+            return out;
+        }
+        SpinHamiltonian::try_from(spin_operator).expect("Error in conversion from SpinOperator to
+SpinHamiltonian, despite the internal check that the HermitianFermionProduct in the jordan-wigner
+transform is hermitian.")
+    }
+}
+
+fn _lowering_operator(i: &usize) -> SpinOperator {
+    let mut out = SpinOperator::new();
+    out.add_operator_product(PauliProduct::new().x(*i), CalculatorComplex::new(0.5, 0.0))
+        .unwrap();
+    out.add_operator_product(PauliProduct::new().y(*i), CalculatorComplex::new(0.0, -0.5))
+        .unwrap();
```

Comment:
I understand that these unwraps should be safe, but they should still be avoided in the code. I'll expand on this in a more general comment